core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -725,6 +728,21 @@ private static SortOrder freshSortOrder(int orderId, Schema schema, SortOrder so
     return builder.build();
   }
 
+  private Map<String, SnapshotRef> validateAndCompleteRefs(Map<String, SnapshotRef> inputRefs) {
```

Comment:
looks like we are just validating refs, should this method name be updated?

Reply:
Ah yes. There is no "completion" as such, since we're always assuming that main exists. I think it could be worth at the end of validating the references in a loop, we perform one final validation that main reference exists in the metadata. Otherwise we fail. Regardless, it is still validation so just "validateRefs" should be good.

I was going to do this in this PR but that breaks a bunch of tests which rely on tableMetadata as the main reference isn't defined in their metadata. Once we get some more clarity though I can add that validation and update the tests. Thoughts?